src/Octoliner.cpp

```diff
@@ -10,292 +10,114 @@
  */
 
 #include "Octoliner.h"
-#include <Wire.h>
 
-void Octoliner::begin(uint8_t brightness) {
-    Wire.begin();
-    pwmFreq(30000);
-    analogWrite(0, brightness);
-    _value = 0;
+Octoliner::Octoliner(uint8_t i2cAddress)
+    : GpioExpander(i2cAddress) {
+    _previousValue = 0.;
 }
 
-void Octoliner::begin(void) {
-    Wire.begin();
-    pwmFreq(30000);
-    analogWrite(0, 200);
-    _value = 0;
+void Octoliner::begin() {
+    Octoliner::begin(&Wire);
 }
 
-void Octoliner::writeCmdPin(IOcommand command, uint8_t pin, bool sendStop) {
-    Wire.beginTransmission(_i2caddress);
-    Wire.write((uint8_t)command);
-    Wire.write(pin);
-    Wire.endTransmission(sendStop);
+void Octoliner::begin(TwoWire* wire) {
+    wire->begin();
+    GpioExpander::begin(wire);
+    GpioExpander::pwmFreq(8000); // ~ 250 different pwm values
```

Comment:
```suggestion
    GpioExpander::pwmFreq(8000); // ~ 250 pwm levels
```

…and why we ever need to touch the PWM settings?

Reply:
Default frequency in stm32f030 firmware is 1kHz. It is good for servo driving, but for sensitivity level set it is slow - will be more noise.